reference/configuration/framework.rst

```diff
@@ -169,39 +293,431 @@ see :doc:`/cookbook/request/load_balancer_reverse_proxy`.
+                <!-- ... -->
+            </framework>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/config.php
+        $container->loadFromExtension('framework', array(
+            'trusted_hosts' => array('acme.com', 'acme.org'),
+        ));
+
+Hosts can also be configured using regular expressions, which make it easier to
+respond to any subdomain:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/config.yml
+        framework:
+            trusted_hosts:  ['.*\.?acme.com$', '.*\.?acme.org$']
+
+    .. code-block:: xml
+
+        <!-- app/config/config.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <framework:config>
+                <framework:trusted-host>.*\.?acme.com$</framework:trusted-host>
+                <framework:trusted-host>.*\.?acme.org$</framework:trusted-host>
+                <!-- ... -->
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/config.php
+        $container->loadFromExtension('framework', array(
+            'trusted_hosts' => array('.*\.?acme.com$', '.*\.?acme.org$'),
+        ));
+
+In addition, you can also set the trusted hosts in the front controller using
+the ``Request::setTrustedHosts()`` method::
+
+    // web/app.php
+    Request::setTrustedHosts(array('.*\.?acme.com$', '.*\.?acme.org$'));
+
+The default value for this option is an empty array, meaning that the application
+can respond to any given host.
+
+.. seealso::
+
+    Read more about this in the `Security Advisory Blog post`_.
+
+.. _reference-framework-form:
+
+form
+~~~~
+
+.. _reference-form-enabled:
+
+enabled
+.......
+
+**type**: ``boolean`` **default**: ``false``
+
+Whether to enable the form services or not in the service container.
+
+This option will automatically be set to ``true`` when one of the child
+settings is configured.
+
+.. note::
+
+    This will automatically enable the validation.
+
+.. seealso::
+
+    For more details, see :doc:`/book/forms`.
+
+csrf_protection
+...............
+
+.. seealso::
+
+    For more information about CSRF protection in forms, see :ref:`forms-csrf`.
+
+.. _reference-form-csrf_protection-enabled:
+
+enabled
+"""""""
+
+**type**: ``boolean`` **default**: ``true``
+
+Whether to enable the CSRF support in forms or not. Setting this option to
+``true`` requires the `secret`_ to be set and `session`_ to be enabled. It
+becomes the service container parameter named ``form.type_extension.csrf.enabled``.
```

Comment:
I think the parameter is an internal detail - i.e. not an API detail that we'd protect with BC. And I can't think of why this would be useful for the user anyways - so safe to remove.